### ISSUE TITLE
`compareJSONString` does not detect changes on Linux

### ISSUE THREAD

**Issue by norio-nomura** [jpsim/SourceKitten#384]
It seems comparing nested `NSDictionary` on Linux does not working at all since Swift 3.0.2. 😞 

It works as expected on macOS:
```terminal.sh-session
➜  18:33:37 git:(master) ✗ swift
swift_oss_helper command enabled.
Welcome to Apple Swift version 3.1 (swiftlang-802.0.53 clang-802.0.42). Type :help for assistance.
  1> import Foundation 
  2. NSDictionary(dictionary: ["key.entities": [["key": 0]]]) == NSDictionary(dictionary: ["key.entities": [["key": 1]]])
$R0: Bool = false
  3>  ^D
```

It does not work as expected on Linux.
Swift 3.0.2:
```terminal.sh-session
$ docker run --privileged -it -v `pwd`:`pwd` -w `pwd` --rm norionomura/swift:302 swift -I /usr/lib/swift/clang/include
Welcome to Swift version 3.0.2 (swift-3.0.2-RELEASE). Type :help for assistance.
  1> import Foundation 
  2. NSDictionary(dictionary: ["key.entities": [["key": 0]]]) == NSDictionary(dictionary: ["key.entities": [["key": 1]]])
expression failed to parse, unknown error
  1> import Foundation 
  2. NSDictionary(dictionary: ["key.entities": [["key": 0]]]) == NSDictionary(dictionary: ["key.entities": [["key": 1]]])
Shared Swift state for repl_swift has developed fatal errors and is being discarded.
REPL definitions and persistent names/types will be lost.
$R0: Bool = true
  3> ^D
```
Swift 3.1:
```terminal.sh-session
$ docker run --privileged -it -v `pwd`:`pwd` -w `pwd` --rm norionomura/swift:31 swift -I /usr/lib/swift/clang/include
Welcome to Swift version 3.1 (swift-3.1-RELEASE). Type :help for assistance.
  1> import Foundation 
  2. NSDictionary(dictionary: ["key.entities": [["key": 0]]]) == NSDictionary(dictionary: ["key.entities": [["key": 1]]])
$R0: Bool = true
  3> ^D
```
Swift 3.1.1:
```terminal.sh-session
$ docker run --privileged -it -v `pwd`:`pwd` -w `pwd` --rm norionomura/swift:311 swift -I /usr/lib/swift/clang/include
Welcome to Swift version 3.1.1 (swift-3.1.1-RELEASE). Type :help for assistance.
  1> import Foundation 
  2. NSDictionary(dictionary: ["key.entities": [["key": 0]]]) == NSDictionary(dictionary: ["key.entities": [["key": 1]]])
$R0: Bool = true
  3> ^D
```
Swift 4.0-DEVELOPMENT-SNAPSHOT-2017-05-15-a:
```terminal.sh-session
$ docker run --privileged -it -v `pwd`:`pwd` -w `pwd` --rm norionomura/swift:4020170515a swift -I /usr/lib/swift/clang/include
Welcome to Swift version 4.0-dev (LLVM f175cad503, Clang 51f7aba0dc, Swift 79b6334dd8). Type :help for assistance.
  1> import Foundation 
  2. NSDictionary(dictionary: ["key.entities": [["key": 0]]]) == NSDictionary(dictionary: ["key.entities": [["key": 1]]])
$R0: Bool = true
  3> ^D
```

**Comment by norio-nomura** [comment]
filed: [SR-4903](https://bugs.swift.org/browse/SR-4903)

**Comment by norio-nomura** [comment]
This issue exists since https://github.com/apple/swift-corelibs-foundation/commit/0f2f4f81a17a8f3490395a804afab52a9e6d798b.
`swift-DEVELOPMENT-SNAPSHOT-2016-08-18-a` and `swift-3.0-GM-CANDIDATE` later.

**Comment by norio-nomura** [comment]
This issue should have been fixed since followings:
`swift-DEVELOPMENT-SNAPSHOT-2017-05-24-a`
`swift-3.1-DEVELOPMENT-SNAPSHOT-2017-05-25`
`swift-4.0-DEVELOPMENT-SNAPSHOT-2017-05-24-a`

**Comment by jpsim** [comment]
Yay. Thanks for filing the bug & fix!

**Comment by norio-nomura** [comment]
This issue has been fixed on Swift 4.0.
Swift 3.1.x with this fixed will not be released.
